### PR TITLE
rename blacklist to deny list

### PR DESCRIPTION
**Pull request by andymckay**
- supports #935
- renames black list to deny list etc, a bit more complicated than the other one
- doesn't deal with JS or the use of blacklist in the validator yet

### CODE REVIEW

File: src/olympia/migrations/907-rename-blacklist.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+RENAME TABLE users_blacklistedname TO users_denied_name;
```
Comment (tofumatt) [inline src/olympia/migrations/907-rename-blacklist.sql]: That's even more readable as a bonus! 👍

File: src/olympia/conf/prod/settings.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 EMAIL_BACKEND = EMAIL_URL['EMAIL_BACKEND']
 EMAIL_HOST_USER = EMAIL_URL['EMAIL_HOST_USER']
 EMAIL_HOST_PASSWORD = EMAIL_URL['EMAIL_HOST_PASSWORD']
-EMAIL_BLACKLIST = env.list('EMAIL_BLACKLIST')
+EMAIL_DENY_LIST = env.list('EMAIL_DENY_LIST')
```
Comment (EnTeQuAk) [inline src/olympia/conf/prod/settings.py]: This needs to be changed by ops first before we can merge this. We already ran into problems of `-dev` got broken. cc @bqbn on this.

**Comment by muffinresearch** [comment]
@bqbn I've filed https://bugzilla.mozilla.org/show_bug.cgi?id=1314584 for the env var.

File: src/olympia/conf/prod/settings.py (hunk 2 above)
Comment (bqbn) [inline src/olympia/conf/prod/settings.py]: Added EMAIL_DENY_LIST to our env settings file for all 3 envs.

**Comment by EnTeQuAk** [comment]
thanks @bqbn 

@andymckay let's :shipit: 😃

**Comment by muffinresearch** [comment]
Once this is landed please merge https://github.com/mozilla/addons-server/pull/3908 also.

**Pull request merged**